Add a description of '--internal' to the docs (#4931)
docker/cli: docs/reference/commandline/network_create.md

### <a name="internal"></a> Network internal mode (--internal)

> Containers on an internal network may communicate between each other, but are
> precluded from communicating with any networks the host has access to (LAN or
> WAN) as no default route is configured, and firewall rules are set up to drop
> all outgoing traffic. Communication with the gateway IP address (and thus
> appropriately configured host services) is possible, and the host may
> communicate with any container IP directly.

Reviewer: Can we make the first sentence shorter and easier to read for non-native readers? For example:

```suggestion
Containers on an internal network may communicate between each other,
but not on any other network, as no default route is configured,
and firewall rules are set up to drop all outgoing traffic.
Communication with the gateway IP address (and thus
appropriately configured host services) is possible, and the host may
communicate with any container IP directly.
```

Author: Thanks David ... I modified that slightly, hopefully it's ok?
